client/my-sites/checkout/checkout-thank-you/marketplace/use-thank-you-steps.tsx

```diff
@@ -0,0 +1,56 @@
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { ThankYouSteps } from 'calypso/components/thank-you/types';
+import useMarketplaceAdditionalSteps from 'calypso/my-sites/marketplace/pages/marketplace-plugin-install/use-marketplace-additional-steps';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { usePluginsThankYouData } from './use-plugins-thank-you-data';
+import { useThemesThankYouData } from './use-themes-thank-you-data';
+import { hasMultipleProductTypes } from './utils';
+
+export function useThankYouSteps( {
+	pluginSlugs,
+	themeSlugs,
+}: {
+	pluginSlugs: Array< string >;
+	themeSlugs: Array< string >;
+} ): ThankYouSteps {
+	const translate = useTranslate();
+	const siteId = useSelector( getSelectedSiteId );
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+
+	const multipleProductTypes = hasMultipleProductTypes( [ pluginSlugs, themeSlugs ] );
+	const additionalSteps = useMarketplaceAdditionalSteps();
+	const randomElement = ( arr: string[] ) => arr[ Math.floor( Math.random() * arr.length ) ];
+
+	const [ , , , pluginsProgressbarSteps ] = usePluginsThankYouData( pluginSlugs );
+	const [ , , , themesProgressbarSteps ] = useThemesThankYouData( themeSlugs );
```

Comment:
I would pass `pluginsProgressbarSteps` and `themesProgressbarSteps` as parameters from the `MarketplaceThankYou` component, in order to avoid retriggering some effects when we call it here.

Reply:
Great point @WBerredo, I've added these as paramters 👍